src/expr/src/linear.rs

```diff
@@ -896,6 +896,33 @@ impl MapFilterProject {
         while prev_size.map(|p| self_size < p).unwrap_or(true) {
             // Lock in current size.
             prev_size = Some(self_size);
+
+            // We have an annoying pattern of mapping literals that already exist as columns (by filters).
+            // Try to identify this pattern, of a map that introduces an expression equated to a prior column,
+            // and then replace the mapped expression by a column reference.
```

Comment:
Could you mention `LiteralLifting` in the comment, and also maybe include one of the diffs? E.g.:
```
- Project (#0, #2)
-   Filter (#1 = 1)
-     Map (1)
-       Get l0
+ Filter (#1 = 1)
+   Get l0
```